src/Commands/MissingDerivatives.php

```diff
@@ -0,0 +1,189 @@
+
+  /**
+   * Identify any objects with missing derivatives.
+   *
+   * @command islandora_drush_utils:missing-derivatives
+   * @aliases islandora_drush_utils:md,idu:md
+   * @usage drush islandora_drush_utils:missing-derivatives --verbose
+   *
+   * @islandora-drush-utils-user-wrap
+   */
+  public function update() {
+    $node_count = $this->getBaseQuery()->count()->execute();
+
+    if ($node_count) {
+      $batch = [
+        'title' => $this->t('Reviewing @node_count object(s) for missing derivatives..',
+          [
+            '@node_count' => $node_count,
+          ]
+        ),
+        'operations' => [
+          [
+            [$this, 'missingDerivatives'],
+            [],
+          ],
+        ],
+      ];
+      drush_op('batch_set', $batch);
+      drush_op('drush_backend_batch_process');
+    }
+    else {
+      $this->logger->log(
+        'info',
+        $this->t(
+          'No nodes of type islandora_object found. Exiting without further processing.'
+        )
+      );
+    }
+
+  }
+
+  /**
+   * Helper to get the base query to be used.
+   *
+   * @return \Drupal\Core\Entity\Query\QueryInterface
+   *   The query to be run.
+   */
+  protected function getBaseQuery() {
+    // Get all nodes relevant.
+    $base_query = $this->entityTypeManager->getStorage('node')
+      ->getQuery()
+      ->condition('type', 'islandora_object');
+    return $base_query;
+  }
+
+  /**
+   * Batch for examining nodes for missing derivatives.
+   *
+   * @param array|\DrushBatchContext $context
+   *   Batch context.
+   */
+  public function missingDerivatives(&$context) {
+    $sandbox =& $context['sandbox'];
+    $base_query = $this->getBaseQuery();
+
+    if (!isset($sandbox['total'])) {
+      $count_query = clone $base_query;
+      $sandbox['total'] = $count_query->count()->execute();
+      if ($sandbox['total'] === 0) {
+        $context['message'] = $this->t('Batch empty.');
+        $context['finished'] = 1;
+        return;
+      }
+      $sandbox['completed'] = 0;
+      $sandbox['last_nid'] = FALSE;
+    }
+
+    if ($sandbox['last_nid']) {
+      $base_query->condition('nid', $sandbox['last_nid'], '>');
+    }
+
+    $base_query->sort('nid');
+    $base_query->range(0, 10);
+
+    foreach ($base_query->execute() as $result) {
+      try {
+        $sandbox['last_nid'] = $result;
+        $node = $this->entityTypeManager->getStorage('node')->load($result);
+        if (!$node) {
+          $this->logger->debug(
+            "Failed to load node {node}; skipping.\n", [
+              'node' => $result,
+            ]
+          );
+          continue;
+        }
+        $context['message'] = dt(
+          "Examining derivatives for node id: @node.\n", [
+            '@node' => $node->id(),
+          ]
+        );
+        $derivative_review = $this->examiner->examine($node);
+        if ($derivative_review) {
+          foreach ($derivative_review as $derivative_message) {
+            $context['message'] = dt(
```

Comment:
Know it's unclear on the ticket but what is the expected output for this? My thought was this would be used to feed the follow up command in some form which probably needs a more parseable style of output.

Perhaps this isn't a concern at this point and maybe it becomes another flag/option as part of the follow-up ticket.